create_task_files.go

```diff
@@ -0,0 +1,106 @@
+package main
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"path"
+	"strings"
+)
+
+func createDirectory(dir string) error {
+	// Create new directory
+	if err := os.Mkdir(dir, 0755); err != nil {
+		return err
+	}
+	return nil
+}
+
+func createFile(file string) error {
+	if _, err := os.Create(file); err != nil {
+		return err
+	}
+	return nil
+}
+
+// Checks that a given task is as expected
+func createTaskFiles(t task) (checkResult, error) {
+	// Checks if in a git repo
+	out, err := exec.Command("git", "rev-parse", "--show-toplevel").Output()
+	if err != nil {
+		message := fmt.Sprintf("Current directory does not seem to be a git repository.\n")
+		return checkResult{Passed: false, Message: message}, nil
+	}
+
+	// Checks for the specified repo
+	if t.GithubRepo != "" {
+		// Obtains the name of current repo
+		repoPath := strings.Split(string(out), "/")
+		depth := len(repoPath) - 1
+		repo := strings.TrimRight(repoPath[depth], "\n")
+
+		// Determines if run from correct repository
+		if repo != t.GithubRepo {
+			message := fmt.Sprintf("No remote pointing to a \"%s\" Github repository; make sure to run this application from your project's directory.", t.GithubRepo)
+			return checkResult{Passed: false, Message: message}, nil
+		}
+
+		// Obtains the current directory name
+		currPath, err := os.Getwd()
+		if err != nil {
+			message := fmt.Sprintf("Error: %s\n", err)
+			return checkResult{Passed: false, Message: message}, nil
+		}
+
+		currDir := strings.Split(currPath, "/")
+		depth = len(currDir) - 1
+		dir := strings.TrimRight(currDir[depth], "\n")
+
+		// Checks if in the root of the repo
```

Comment:
Shouldn't we just use the output of `git rev-parse --show-top-level`, to just append the folder and file names to it directly, to create them?